wcs-functions.php

```diff
@@ -114,13 +113,19 @@ function wcs_create_subscription( $args = array() ) {
 		'start_date'         => $default_start_date,
 		'date_created'       => $now,
 		'created_via'        => ( ! empty( $order ) ) ? wcs_get_objects_property( $order, 'created_via' ) : '',
-		'order_version'      => ( ! empty( $order ) ) ? wcs_get_objects_property( $order, 'version' ) : WC_VERSION,
 		'currency'           => ( ! empty( $order ) ) ? wcs_get_objects_property( $order, 'currency' ) : get_woocommerce_currency(),
 		'prices_include_tax' => ( ! empty( $order ) ) ? ( ( wcs_get_objects_property( $order, 'prices_include_tax' ) ) ? 'yes' : 'no' ) : get_option( 'woocommerce_prices_include_tax' ), // we don't use wc_prices_include_tax() here because WC doesn't use it in wc_create_order(), not 100% sure why it doesn't also check the taxes are enabled, but there could forseeably be a reason
 	);
 
-	$args              = wp_parse_args( $args, $default_args );
-	$subscription_data = array();
+	if ( isset( $args['order_version'] ) ) {
+		wcs_deprecated_argument( __FUNCTION__, '2.0', 'The "order_version" argument is no longer changeable.' );
+	}
+
+	$args = wp_parse_args( $args, $default_args );
+
+	if ( ! empty( $args['status'] ) && ! array_key_exists( 'wc-' . $args['status'], wcs_get_subscription_statuses() ) ) {
```

Comment:
Nitpick: It might be nice to add a comment like `// Validate the status arg.` for consistency with comments below.

Reply:
Resolved in eede953fbe6ed9f8bcab26559db60ce0c16bff89